src/org/traccar/reports/ReportUtils.java

```diff
@@ -237,8 +239,11 @@ private static StopReport calculateStop(ArrayList<Position> positions, int start
 
     private static boolean isMoving(ArrayList<Position> positions, int index,
             TripsConfig tripsConfig, double speedThreshold) {
-        if (tripsConfig.getMinimalNoDataDuration() > 0 && index < positions.size() - 1
+        if (tripsConfig.getMinimalNoDataDuration() > 0 && (index < positions.size() - 1
                 && positions.get(index + 1).getFixTime().getTime() - positions.get(index).getFixTime().getTime()
+                >= tripsConfig.getMinimalNoDataDuration())
+                || index > 0
+                && positions.get(index).getFixTime().getTime() - positions.get(index - 1).getFixTime().getTime()
```

Comment:
Can you please split parts of this expression into variables. It's very hard to understand what's going on.